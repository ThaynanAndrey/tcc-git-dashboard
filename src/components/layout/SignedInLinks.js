import React from 'react';
import { NavLink } from 'react-router-dom';

/**
 * Stateless Component with links to Navbar when user is authenticated.
 * 
 * @author Thaynan Nunes
 */
const SignedInLinks = (props) => {
    return (
        <ul className="right hide-on-med-and-down">
            <li><NavLink to='/projetos'>Projetos</NavLink></li>
            <li><NavLink to='/login' onClick={props.signOut}>
                Sair
            </NavLink></li>
        </ul>
    );
};

export default SignedInLinks;