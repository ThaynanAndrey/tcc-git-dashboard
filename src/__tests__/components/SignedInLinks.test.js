import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import SignedInLinks from '../../components/layout/SignedInLinks';

describe('Test SignedInLinks', () => {
  it('renders correctly a component SignedInLinks with classes right hide-on-med-and-down', () => {
    const tree = shallow(
      <SignedInLinks />
    );
    
    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find(".right.hide-on-med-and-down").exists()).toBeTruthy();
  });
});