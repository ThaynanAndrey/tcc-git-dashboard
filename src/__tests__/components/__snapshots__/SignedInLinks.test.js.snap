// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SignedInLinks renders correctly a component SignedInLinks with classes right hide-on-med-and-down 1`] = `
<ul
  className="right hide-on-med-and-down"
>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/projetos"
    >
      Projetos
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/login"
    >
      Sair
    </NavLink>
  </li>
</ul>
`;
